CMakeLists.txt

```diff
@@ -325,6 +351,11 @@ IF (BUILD_TESTING)
   function(AddTest) # function for UT
     cmake_parse_arguments(UT "DYN" "TARGET" "LIBS;DYN_LIBS;STATIC_LIBS;SOURCES;DEPENDS" ${ARGN})
     add_executable(${UT_TARGET} ${UT_SOURCES})
+
+    if(ENABLE_COVERAGE)
+      add_coverage(${UT_TARGET})
```

Comment:
Can we add compilation flags manually, preventing `add_coverage` for each target?

Reply:
Yes, but dependency error may occur easily, so we may add it later when we have dealt with coverage dependencies.